test/python/transpiler/test_swap_strategy_router.py

```diff
@@ -562,9 +562,31 @@ def test_edge_coloring(self, edge_coloring):
 
         self.assertEqual(pm_.run(circ), expected)
 
+    def test_layout(self):
+        """Test that circuit layout permutations are properly tracked in the pass property
+        set and returned with the routed circuit."""
+
+        # We use the same scenario as the QAOA test above
+        mixer = QuantumCircuit(4)
+        for idx in range(4):
+            mixer.ry(-idx, idx)
+
+        op = SparsePauliOp.from_list([("IZZI", 1), ("ZIIZ", 2), ("ZIZI", 3)])
+        circ = QAOAAnsatz(op, reps=2, mixer_operator=mixer)
+
+        expected_initial_layout = Layout.generate_trivial_layout(*circ.qregs)
+        expected_final_index_layout = [3, 1, 2, 0]
+        expected_final_layout = Layout.from_intlist(expected_final_index_layout, *circ.qregs)
+
+        swapped = self.pm_.run(circ.decompose())
+
+        self.assertEqual(swapped.layout.initial_layout, expected_initial_layout)
+        self.assertEqual(swapped.layout.final_layout, expected_final_layout)
+        self.assertEqual(swapped.layout.final_index_layout(), expected_final_index_layout)
```

Comment:
There's no layout pass in your `PassManager` here, so I wouldn't expect `initial_layout` to be set by anything in the pass manager.

fwiw, you don't need to test `TranspileLayout.initial_layout` or `TranspileLayout.final_layout` - we recommend not using either of those legacy attributes, but only using the methods on `TranspileLayout` following #10835.  In this case, since the swap-strategy thing is just a router, the only method I'd really expect to return a non-`None` value is `TranspileLayout.routing_permutation`.